Justfile

```diff
@@ -14,193 +13,171 @@ MAGENTA := '\033[35m'
 CYAN := '\033[36m'
 WHITE := '\033[37m'
 GRAY := '\033[90m'
-
-# Background colors
-BG_BLACK := '\033[40m'
 BG_RED := '\033[41m'
-BG_GREEN := '\033[42m'
-BG_YELLOW := '\033[43m'
-BG_BLUE := '\033[44m'
-BG_MAGENTA := '\033[45m'
-BG_CYAN := '\033[46m'
-BG_WHITE := '\033[47m'
-
-# Text styles
 BOLD := '\033[1m'
-DIM := '\033[2m'
-ITALIC := '\033[3m'
-UNDERLINE := '\033[4m'
-
-# Reset all styles
 NC := '\033[0m'
-
-# Display a symbol
 CHECK := "$(GREEN)✓$(NC)"
 CROSS := "$(RED)✗$(NC)"
 DASH := "$(GRAY)-$(NC)"
 
-
-# List all available recipes
 @default:
     just --list --unsorted
 
-# Check if required tools are installed
 @check-deps:
     @#!/usr/bin/env sh
     if ! command -v uv >/dev/null 2>&1; then echo "uv is not installed"; exit 1; fi
     if ! command -v python3 >/dev/null 2>&1; then echo "python3 is not installed"; exit 1; fi
     if ! command -v just >/dev/null 2>&1; then echo "just is not installed"; exit 1; fi
-    if ! command -v pre-commit >/dev/null 2>&1; then echo "{{YELLOW}}WARNING: pre-commit is not installed{{NC}}"; fi
+    if ! command -v pre-commit >/dev/null 2>&1; then echo "$(YELLOW)WARNING: pre-commit is not installed$(NC)"; fi
```

Comment:
Change believe $(NC) is the wrong bash syntax should be {{NC}} which is justfile syntax please verify.